Add Cypress artifacts documentation (#201)
transcom/mymove-docs: docs/backend/testing/run-e2e-tests.md

```sh
yarn cypress run --spec cypress/integration/path/to/file.jsx
```

## Automated tests artifacts from continuous integration

:::note External documentation call-out
The intention of this documentation is to not repeat the official docs. At the
time of this writing, the documentation being referenced contains an _Artifacts

> overview_ section is the main area that shows where Artifacts are stored in the

Reviewer: ```suggestion
overview_ section, which shows where Artifacts are stored in the
```
